src/DependencyInjection/Configuration.php

```diff
@@ -39,13 +39,49 @@ public function getConfigTreeBuilder(): TreeBuilder
+                                ->scalarNode('public_url')->info('URL of the hub\'s public endpoint')->example('https://demo.mercure.rocks/.well-known/mercure')->defaultNull()->end()
+                                ->arrayNode('jwt')
+                                    ->beforeNormalization()
+                                        ->ifString()
+                                        ->then(static function (string $token): array {
+                                            return [
+                                                'value' => $token,
+                                            ];
+                                        })
+                                    ->end()
+                                    ->info('JSON Web Token configuration.')
+                                        ->children()
+                                            ->scalarNode('value')->info('JSON Web Token to use to publish to this hub.')->end()
+                                            ->scalarNode('provider')->info('The ID of a service to call to provide the JSON Web Token.')->end()
+                                            ->scalarNode('factory')->info('The ID of a service to call to create the JSON Web Token.')->end()
+                                            ->arrayNode('publish')
+                                                ->beforeNormalization()->castToArray()->end()
+                                                ->scalarPrototype()->end()
+                                                ->info('A list of topics to allow publishing to when using the given factory to generate the JWT.')
+                                            ->end()
+                                            ->arrayNode('subscribe')
+                                                ->beforeNormalization()->castToArray()->end()
+                                                ->scalarPrototype()->end()
+                                                ->info('A list of topics to allow subscribing to when using the given factory to generate the JWT.')
+                                            ->end()
+                                            ->scalarNode('secret')->info('The JWT Secret to use.')->example('!ChangeMe!')->end()
+                                        ->end()
+                                ->end()
+                                ->scalarNode('jwt_provider')
+                                    ->info('The ID of a service to call to generate the JSON Web Token.')
+                                    ->setDeprecated('symfony/mercure-bundle', '0.3', 'The child node "%node%" at path "%path%" is deprecated, use "jwt.provider" instead.')
+                                ->end()
                                 ->scalarNode('bus')->info('Name of the Messenger bus where the handler for this hub must be registered. Default to the default bus if Messenger is enabled.')->end()
```

Comment:
I suggest deprecating this one (maybe in a follow-up PR).

Reply:
i think we should keep this for later, if we are going to deprecate this feature, then we should remove the messenger handler in the component ( or deprecate it )

Reply:
Yes deprecate it

Reply:
What's the motivation?

Reply:
because sending the request to mercure hub directly is probably cheaper ( resource wise ) then having to send it to rabbitmq/redis/database just to send it to mercure hub in a the worker.

it's useful to run things by messenger when the will result in multiple I/O interactions, but replacing only 1 by another is not helpful.

Reply:
agreed, thanks.